result.go

```diff
@@ -69,3 +71,7 @@ func (res *snowflakeResult) waitForAsyncExecStatus() error {
 	}
 	return nil
 }
+
+func (res *snowflakeResult) Monitoring() *QueryMonitoringData {
```

Comment:
GetStatus and Monitoring both use monitoring/.. REST API. These two can share some code. And we prefer fetching monitoring data only when status or monitoring is needed, and we should not return the same saved monitoring when Monitoring is called. What do you think?

Reply:
> These two can share some code.

Sounds good. How would you like to handle the data structures (`QueryMonitoringData` and `retStatus`)? I think we should keep them separate since they represent different views on the same underlying data (the only field they have in common is `Status`), but I can also merge them into a single struct if you'd prefer.

> we prefer fetching monitoring data only when status or monitoring is needed

Sounds good.

> we should not return the same saved monitoring when Monitoring is called

To make sure I understand, it sounds like you'd prefer each call to `Monitoring` to re-fetch the monitoring data. Is there a reason you don't want to cache the monitoring information and reuse it? Do we expect the monitoring data for a given query to change between calls to `Monitoring`? All of the fields in `QueryMonitoringData` look like they should be fixed once a query has been run.

Reply:
As a follow-on question, how would we actually implement the re-fetching? At the time that the client calls `.Monitoring()` on a `SnowflakeResult`, we no longer have access to the `snowflakeConn` object that we need to call `FuncGet()`.

Reply:
After looking at this more closely, I'm not sure there's a good way to expose the monitoring information, since it's fundamentally not part of the golang sql interface. If we return the monitoring data as part of the result (as in the current implementation of this PR), the client needs a way to get access to the query result as a `SnowflakeResult` rather than a generic `Rows` object. If we expose monitoring data via a top-level method on `snowflakeConn`, the client needs a way to get access to the `snowflakeConn` as a `snowflakeConn` rather than a generic `DB` object. In either case, we're exposing something that isn't currently part of the public interface.